sympy/core/mul.py

```diff
@@ -1250,6 +1250,56 @@ def _eval_is_zero(self):
         return zero
 
     def _eval_is_integer(self):
+        if len(self.args) == 2:
```

Comment:
What about deleting all of this and replacing
```
elif d == S(2):
                return n.is_even
```
with
```
if n.is_prime and d.is_prime:
    return False
m = Mod(n, d).is_integer
if m is not None:
    return m
```

Reply:
umm, do you mean deleting the whole `if len(self.args) ==2:` block ?

Reply:
yes - see what fails after making the change. And if nothing fails then we have a simpler solution to the problem.

Reply:
A couple of em fail ...
here's one 
```python3
  File "/Users/mosespaul/Desktop/Coding/GSoC/sympy2/sympy/functions/combinatorial/tests/test_comb_factorials.py", line 341, in test_factorial2
    assert factorial2(tfo).is_positive is None
AssertionError
```

Reply:
Ive a question, `4 * 1/3`
you'll get something like `Mod(4, 3) == 1 ` which is an integer
so 
```python3
>>> Mul(4, Rational(1,3), evaluate=False).is_integer #this is False
True
```

Reply:
rust-flaking-off-my-brain-as-I-say, "yes"

Reply:
🤣 I made the correct changes tho , I'm testing them, will push em.